specs/xaml-resource-references-tracing-spec.md

```diff
@@ -0,0 +1,189 @@
+  Finished searching dictionary 'ms-appx:///Controls/ControlExample.xaml'.
+  Searching dictionary 'Framework-defined colors' for resource with key 'OutputTextBlockStyl'.
+  Finished searching dictionary 'Framework-defined colors'.
+  Searching dictionary 'Framework ThemeResources.xbf' for resource with key 'OutputTextBlockStyl'.
+    Searching theme dictionary (active theme: 'Light') for resource with key 'OutputTextBlockStyl'.
+      Searching dictionary '<anonymous dictionary>' for resource with key 'OutputTextBlockStyl'.
+      Finished searching dictionary '<anonymous dictionary>'.
+    Finished searching theme dictionary (active theme: 'Light').
+  Finished searching dictionary 'Framework ThemeResources.xbf'.
+  Searching dictionary 'ms-appx:///App.xaml' for resource with key 'OutputTextBlockStyl'.
+    Searching merged dictionary with index '1' for resource with key 'OutputTextBlockStyl'.
+      Searching dictionary 'ms-appx:///Microsoft.UI.Xaml/Themes/themeresources.xaml' for resource with key 'OutputTextBlockStyl'.
+        Searching theme dictionary (active theme: 'Light') for resource with key 'OutputTextBlockStyl'.
+          Searching dictionary 'ms-appx:///Microsoft.UI.Xaml/Themes/themeresources.xaml' for resource with key 'OutputTextBlockStyl'.
+          Finished searching dictionary 'ms-appx:///Microsoft.UI.Xaml/Themes/themeresources.xaml'.
+        Finished searching theme dictionary (active theme: 'Light').
+      Finished searching dictionary 'ms-appx:///Microsoft.UI.Xaml/Themes/themeresources.xaml'.
+    Finished searching merged dictionary with index '1'.
+    Searching merged dictionary with index '0' for resource with key 'OutputTextBlockStyl'.
+      Searching dictionary 'ms-appx:///ItemTemplates.xaml' for resource with key 'OutputTextBlockStyl'.
+      Finished searching dictionary 'ms-appx:///ItemTemplates.xaml'.
+    Finished searching merged dictionary with index '0'.
+    Searching theme dictionary (active theme: 'Light') for resource with key 'OutputTextBlockStyl'.
+      Searching dictionary 'ms-appx:///App.xaml' for resource with key 'OutputTextBlockStyl'.
+      Finished searching dictionary 'ms-appx:///App.xaml'.
+    Finished searching theme dictionary (active theme: 'Light').
+  Finished searching dictionary 'ms-appx:///App.xaml'.
+Finished search for resource with key 'OutputTextBlockStyl'.
+```
+
+
+## XamlResourceReferenceFailedEventArgs.Message Property
+
+Gets the explanation of the XAML resource reference failure.
+
+```c#
+public string Message { get; }
+```
+
+
+# API Details
+```c#
+namespace Microsoft.UI.Xaml
+{
+  runtimeclass DebugSettings
+  {
+    // existing ...
+
+    Boolean IsXamlResourceReferenceTracingEnabled;
+    event Windows.Foundation.TypedEventHandler<Microsoft.UI.Xaml.DebugSettings,Microsoft.UI.Xaml.XamlResourceReferenceFailedEventArgs> XamlResourceReferenceFailed;
+  };
+
+  runtimeclass XamlResourceReferenceFailedEventArgs
+  {
+    XamlResourceReferenceFailedEventArgs();
```

Comment:
Is the public constructor necessary?

Reply:
No, this isn't expected to be constructed by the user.